core/trino-main/src/test/java/io/trino/execution/TestSetNotNullConstraintConstraintTask.java

```diff
@@ -0,0 +1,186 @@
+                .hasErrorCode(NOT_SUPPORTED)
+                .hasMessageContaining("Column is already not nullable");
+    }
+
+    @Test
+    void testSetNotNullConstraintOnView()
+    {
+        QualifiedObjectName viewName = qualifiedObjectName("existing_view");
+        metadata.createView(testSession, viewName, someView(), ImmutableMap.of(), false);
+
+        assertTrinoExceptionThrownBy(() -> getFutureValue(executeSetNotNullConstraint(asQualifiedName(viewName), identifier("test"), false)))
+                .hasErrorCode(TABLE_NOT_FOUND)
+                .hasMessageContaining("Table '%s' does not exist, but a view with that name exists.", viewName);
+    }
+
+    @Test
+    void testSetNotNullConstraintOnMaterializedView()
+    {
+        QualifiedObjectName materializedViewName = qualifiedObjectName("existing_materialized_view");
+        metadata.createMaterializedView(testSession, QualifiedObjectName.valueOf(materializedViewName.toString()), someMaterializedView(), MATERIALIZED_VIEW_PROPERTIES, false, false);
+
+        assertTrinoExceptionThrownBy(() -> getFutureValue(executeSetNotNullConstraint(asQualifiedName(materializedViewName), identifier("test"), false)))
+                .hasErrorCode(TABLE_NOT_FOUND)
+                .hasMessageContaining("Table '%s' does not exist, but a materialized view with that name exists.", materializedViewName);
+    }
+
+    private ListenableFuture<Void> executeSetNotNullConstraint(QualifiedName table, Identifier column, boolean tableExists)
+    {
+        return new SetNotNullConstraintTask(plannerContext.getMetadata(), new AllowAllAccessControl())
+                .execute(new SetNotNullConstraint(new NodeLocation(1, 1), table, column, tableExists), queryStateMachine, ImmutableList.of(), WarningCollector.NOOP);
+    }
+
+    private static ConnectorTableMetadata simpleTable(QualifiedObjectName tableName)
+    {
+        return new ConnectorTableMetadata(tableName.asSchemaTableName(), ImmutableList.of(nullableColumn("a", BIGINT), nullableColumn("b", BIGINT)));
+    }
+
+    private static ColumnMetadata nullableColumn(String name, Type type)
+    {
+        return ColumnMetadata.builder()
+                .setName(name)
+                .setType(type)
+                .setNullable(true)
+                .build();
+    }
+
+    private static ColumnMetadata notNullColumn(String name, Type type)
+    {
+        return ColumnMetadata.builder()
+                .setName(name)
+                .setType(type)
+                .setNullable(false)
+                .build();
+    }
+
+    private static class MockMetadataWithNotNull
+            extends MockMetadata
+    {
+        public MockMetadataWithNotNull(String catalogName)
+        {
+            super(catalogName);
+        }
+
+        @Override
+        public Set<ConnectorCapabilities> getConnectorCapabilities(Session session, CatalogHandle catalogHandle)
+        {
+            return ImmutableSet.of(NOT_NULL_COLUMN_CONSTRAINT);
```

Comment:
Check for `NOT_NULL_COLUMN_CONSTRAINT` looks missing in `SetNotNullConstraintTask` & `DropNotNullConstraintTask`.